source/common/http/async_client_impl.cc

```diff
@@ -142,17 +142,48 @@ void AsyncStreamImpl::sendTrailers(HeaderMap& trailers) {
 }
 
 void AsyncStreamImpl::closeLocal(bool end_stream) {
+  // TODO(goaway): This assert maybe merits reconsideration. It seems to be saying that we shouldn't
```

Comment:
Yeah I agree, I don't remember the history here. We should look at cleaning this up. cc @htuch @lizan in case they remember any of this history.